src/test/java/io/github/adorjan/loader/PropertiesLoaderTest.java

```diff
@@ -0,0 +1,67 @@
+package io.github.adorjan.loader;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.slf4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class PropertiesLoaderTest {
+
+    private final String KEY = "greeting.message";
+    private final String VALUE = "Hello All from properties file!";
+
+    private PropertiesLoader underTest;
+    @Mock
+    private Properties mockProperties;
+    @Mock
+    private InputStream mockInputStream;
+    @Mock
+    private Logger mockLogger;
+
+    @BeforeMethod
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        underTest = new PropertiesLoader();
+        underTest.setLogger(mockLogger);
+    }
+
+    @Test
+    public void testLoadShouldThrowFileNotFoundExceptionWhenInputStreamIsEmpty() {
+        // GIVEN in setUp
+        // WHEN
+        underTest.load(Optional.empty());
+        // THEN
+        Mockito.verify(mockLogger, Mockito.only()).error(Mockito.anyString(), Mockito.any(FileNotFoundException.class));
```

Comment:
I think verifying with the proper string ("_Properties file not found_") would result a better unit test.